### PR TITLE
Rust: add QL doc annotations to schema

### PR DESCRIPTION
**Pull request by aibaars**
This pull request puts the QL docs with code examples  back into the schema and generate the corresponding QL tests.

### CODE REVIEW

File: rust/schema/annotations.py [hunk 1]
```diff
@@ -0,0 +1,1741 @@
...
+# @rust.doc_test_signature("() -> bool")
+# class _:
+#     """
+#     A `const` block expression. For example:
+#     ```rust
+#     if const { SRC::IS_ZST || DEST::IS_ZST || mem::align_of::<SRC>() != mem::align_of::<DEST>() } {
+#         return false;
+#     }
+#     ```
+#     """
```
Comment (redsun82) [inline rust/schema/annotations.py:183]: why is this commented out? Might be worth a TODO unless it's an oversight

File: rust/schema/annotations.py [hunk 2]
```diff
@@ -0,0 +1,1741 @@
...
+    ```
+    """
+
+
+@annotate(Expr)
+class _:
+    """
+    The base class for expressions.
+    """
+    pass
```
Comment (redsun82) [inline rust/schema/annotations.py:26]: nit: `pass` is not required, and currently the file is not very consistent on it :)

File: rust/schema/annotations.py [hunk 3]
```diff
@@ -0,0 +1,1741 @@
...
+
+
+# @annotate(RepeatExpr)
+# class _:
+#     """
+#     A repeat expression. For example:
+#     ```rust
+#     [1; 10];
+#     ```
+#     """
```
Comment (redsun82) [inline rust/schema/annotations.py:558]: I guess this is what because of what you mentioned on https://github.com/github/codeql-team/issues/3378, right?

Comment (aibaars) [reply rust/schema/annotations.py:558]: Yes, but I can remove this for now.

**Comment by aibaars** [comment]
@redsun82 The QL test generator seems to ignore the function signatures somehow, did we break something in the schema generator?

**Comment by redsun82** [comment]
> @redsun82 The QL test generator seems to ignore the function signatures somehow, did we break something in the schema generator?

seems like it, but also seems like https://github.com/github/codeql/pull/17533 fixes it :slightly_smiling_face:

**Review by redsun82 — APPROVED** [review]
🚀

**Pull request merged**